examples/step-95/step-95.cc

```diff
@@ -0,0 +1,1711 @@
+    const unsigned int quad_index = 0;
+
+    bool is_dg = false;
+  };
+
+  // @sect3{The Jacobi preconditioner}
+  // Assembly is done in the constructor by calling the compute_diagonal()
+  // function of the operator.
+  template <int dim>
+  class JacobiPreconditioner
+  {
+  public:
+    using VectorType = LinearAlgebra::distributed::Vector<double>;
+
+    JacobiPreconditioner(const PoissonOperator<dim> &poisson_operator)
+    {
+      poisson_operator.initialize_dof_vector(inverse_diagonal);
+      poisson_operator.compute_diagonal(inverse_diagonal);
+      for (unsigned int i = 0; i < inverse_diagonal.locally_owned_size(); ++i)
+        {
+          if (std::abs(inverse_diagonal.local_element(i)) > 1.0e-10)
+            inverse_diagonal.local_element(i) =
+              1.0 / inverse_diagonal.local_element(i);
+          else
+            inverse_diagonal.local_element(i) = 1.0;
+        }
+    }
+
+    void vmult(VectorType &dst, const VectorType &src) const
+    {
+      if (PointerComparison::equal(&dst, &src))
+        {
+          dst.scale(inverse_diagonal);
+        }
+      else
+        {
+          for (unsigned int i = 0; i < dst.locally_owned_size(); ++i)
+            dst.local_element(i) =
+              inverse_diagonal.local_element(i) * src.local_element(i);
+        }
+    }
+
+  private:
+    VectorType inverse_diagonal;
+  };
+
+  // @sect3{The PoissonSolver class template}
+  template <int dim>
+  class PoissonSolver
+  {
+    using Number              = double;
+    using VectorizedArrayType = VectorizedArray<Number>;
+    using VectorType          = LinearAlgebra::distributed::Vector<Number>;
+
+    using CellIntegrator =
+      FEEvaluation<dim, -1, 0, 1, Number, VectorizedArrayType>;
+    using GenericCellIntegrator =
+      FEPointEvaluation<1, dim, dim, VectorizedArrayType>;
+
+    static constexpr unsigned int n_lanes = VectorizedArrayType::size();
+
+  public:
+    PoissonSolver();
+
+    void run(bool is_dg, unsigned int fe_degree);
+
+  private:
+    void make_grid();
+
+    void setup_discrete_level_set();
+
+    void distribute_dofs();
+
+    void setup_matrix_free();
+
+    void setup_mapping_data();
+
+    void solve();
+
+    void output_results() const;
+
+    double compute_L2_error() const;
+
+    unsigned int fe_degree;
+
+    const Functions::ConstantFunction<dim> rhs_function;
+
+    parallel::distributed::Triangulation<dim> triangulation;
+
+    ConditionalOStream pcout;
+
+    // MatrixFree object
+    MatrixFree<dim, double> matrix_free;
+
+    PoissonOperator<dim> poisson_operator;
+
+    // We need two separate DoFHandlers. The first manages the DoFs for the
+    // discrete level set function that describes the geometry of the domain.
+    std::unique_ptr<FE_Q<dim>> fe_level_set;
+    DoFHandler<dim>            level_set_dof_handler;
+    VectorType                 level_set;
+
+    // The second DoFHandler manages the DoFs for the solution of the Poisson
+    // equation.
```

Comment:
Also here, try to write larger comment blocks that do not interrupt the code as often.